src/main/java/org/wildfly/security/auth/provider/SimpleMapBackedSecurityRealm.java

```diff
@@ -118,5 +121,17 @@ public CredentialSupport getCredentialSupport(Class<?> credentialType) {
             final Password password = map.get(principal);
             return credentialType.isInstance(password) ? credentialType.cast(password) : null;
         }
+
+        public VerificationResult verifyCredential(final Object credential) {
```

Comment:
We could actually leave the invalid combination in as a valid combination on the basis the realm author may not have implemented that verify method and so wants to claim it verifies nothing.  I think the 'Supports' claim is about what the realm supports and not what is technically possible.